janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLKeyColumnValueStore.java

```diff
@@ -448,15 +447,17 @@ public Row next() {
                 this.index = 0;
             }
             this.index++;
-            lastPagingState = currentResultSet.getExecutionInfo().getPagingState();
+            // Using getPagingStateUnsafe rather than getPagingState, to avoid the md5
+            // compute of the pageState, which can be quite expensive.
+            lastPagingState = currentResultSet.getExecutionInfo().getPagingStateUnsafe();
```

Comment:
How is this related to the index improvements? If it's completely unrelated, then please create a separate PR (and probably also its own issue) for this.

Reply:
It was kind of related. We are using CQL adapter, and we found that the above change is one of the biggest performance improvements while indexing. Before the above change, it was taking ~4-5 hours for ~1M vertexes and after the change, it's about 15mins. The largest time being spent was on `getPagingState` which does md5sum compute of the entire resultset.

Reply:
That is of course a significant improvement, but I would still prefer to handle that outside of this PR as it's a CQL specific improvement and such a significant improvement shouldn't be "hidden" in this PR.

Reply:
Sure @FlorianHockmann. I reverted the change for CQLKeyColumnValueStore, and will create a new PR only for CQL optimization.